app/models/user.rb

```diff
@@ -163,6 +163,10 @@ def internal?
     auth_source.is_a? AuthSourceInternal
   end
 
+  def can_update_username?
+    hidden? or internal? or admin?
```

Comment:
Nitpick, this is control-flow OR, it should be logic OR (http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/)